Sources/src/FromSTMTestBeamData_source.cc

```diff
@@ -0,0 +1,335 @@
+//
+// Source module to read binary files from the STM test beam
+// at gELBE collected in April 2022
+//
+#include <iostream>
+#include <fstream>
+#include <boost/utility.hpp>
+#include <cassert>
+#include <set>
+#include <string>
+#include <vector>
+
+#include "fhiclcpp/types/Name.h"
+#include "art/Framework/IO/Sources/Source.h"
+#include "art/Framework/Core/InputSourceMacros.h"
+#include "art/Framework/IO/Sources/SourceHelper.h"
+#include "art/Framework/Principal/RunPrincipal.h"
+#include "art/Framework/Principal/SubRunPrincipal.h"
+#include "art/Framework/Principal/EventPrincipal.h"
+#include "art/Framework/IO/Sources/put_product_in_principal.h"
+#include "canvas/Persistency/Provenance/Timestamp.h"
+#include "canvas/Persistency/Provenance/RunID.h"
+#include "canvas/Persistency/Provenance/SubRunID.h"
+#include "canvas/Persistency/Provenance/EventID.h"
+#include "canvas/Persistency/Provenance/BranchType.h"
+#include "canvas/Persistency/Provenance/ProductID.h"
+#include "canvas/Persistency/Provenance/canonicalProductName.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Services/Registry/ServiceHandle.h"
+
+#include "Offline/Sources/inc/STMTestBeamHeaders.hh"
+#include "Offline/DataProducts/inc/STMTestBeamEventInfo.hh"
+#include "Offline/RecoDataProducts/inc/STMWaveformDigi.hh"
+
+
+using namespace std;
+
+namespace mu2e {
+  struct Config
+  {
+    using Name = fhicl::Name;
+    using Comment = fhicl::Comment;
+    fhicl::Sequence<std::string> inputFiles{Name("fileNames"),Comment("Input binary file")};
+    fhicl::Atom<unsigned int> maxEvents{Name("maxEvents"), Comment("Max number of events")};
+    fhicl::Atom<int> verbosityLevel{Name("verbosityLevel"), Comment("Verbosity level")};
+
+    // These are used by art and are required.
+    fhicl::Atom<std::string> module_label{Name("module_label"), Comment("Art module label"), ""};
+    fhicl::Atom<std::string> module_type{Name("module_type"), Comment("Art module type"), ""};
+  };
+  typedef fhicl::WrappedTable<Config> Parameters;
+
+
+  //================================================================
+  class STMTestBeamDataDetail : private boost::noncopyable {
+    std::string myModuleLabel_;
+    art::SourceHelper const& pm_;
+    unsigned runNumber_; // from file name
```

Comment:
Write this as   ```unsigned runNumber_ = 0;```  Then it is implicitly initialized in every c'tor.  This gets the boiler plate out of the c'tor, making the c'tor shorter and easier to read.  The same for a lot of other member data.   For data members of type string, it's OK to let them default construct (if that's what you want) and not mention them in the c'tor initializer list.

Reply:
Done in [86c7106](https://github.com/Mu2e/Offline/pull/856/commits/86c7106ce0d9319e26f96d34d4fdbda49c37ab29)